src/sinks/util/adaptive_concurrency/tests.rs

```diff
@@ -382,17 +412,14 @@ async fn run_test(params: TestParams) -> TestResults {
 
     let controller = get_controller().unwrap();
 
-    // This is crude and dumb, but it works, and the tests run fast and
-    // the results are highly repeatable.
-    while stats.lock().expect("Poisoned stats lock").completed < params.requests {
-        time::sleep(Duration::from_millis(1)).await;
-    }
+    is_done.await.expect("Test failed to complete");
```

Comment:
All that for this little payoff, but now these tests should be less dependent on the internal behavior of tokio timers.